sbt-cross/src/main/scala/sbtcross/CrossProject.scala

```diff
@@ -0,0 +1,213 @@
+package sbtcross
+
+import sbtcross._
+
+import scala.language.implicitConversions
+
+import sbt._
+import Keys._
+
+final class CrossProject private[sbtcross] (
+    val id: String,
+    crossType: CrossType,
+    val projects: Map[Platform, Project]
+) {
+
+  def platforms = projects.keySet
+
+  def aggregate(refs: CrossProject*): CrossProject = {
+    requireDependencies(refs.toList)
+
+    val aggregatesByPlatform =
+      refs.toSeq.flatMap(_.projects).groupBy(_._1).mapValues(_.map(_._2))
+
+    mapProjectsByPlatform((platform, project) =>
+      project.aggregate(aggregatesByPlatform(platform).map(p =>
+        p: ProjectReference): _*))
+  }
+
+  def dependsOn(deps: CrossClasspathDependency*): CrossProject = {
+    requireDependencies(deps.toList.map(_.project))
+
+    val dependenciesByPlatform =
+      deps.toSeq
+        .flatMap(dep =>
+          dep.project.projects.map {
+            case (platform, project) =>
+              platform -> ClasspathDependency(project, dep.configuration)
+        })
+        .groupBy(_._1)
+        .mapValues(_.map(_._2))
+
+    mapProjectsByPlatform((platform, project) =>
+      project.dependsOn(dependenciesByPlatform(platform): _*))
+  }
+
+  def configs(cs: Configuration*): CrossProject =
+    transform(_.configs(cs: _*))
+
+  def configureCross(
+      transforms: (CrossProject => CrossProject)*): CrossProject =
+    transforms.foldLeft(this)((p, t) => t(p))
+
+  def configure(transforms: (Project => Project)*): CrossProject =
+    transform(_.configure(transforms: _*))
+
+  @deprecated("use configure", "now")
```

Comment:
oh right, I totally forgot about this one. We introduced because of this comment in Scala.js:

```
// TODO: rename to "configure" when compatibility can be broken (1.0.0)
//       and the existing deprecated "configure" is removed
```